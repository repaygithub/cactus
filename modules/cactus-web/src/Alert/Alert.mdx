---
name: Alert
menu: Components
---

import PropsTable from 'website-src/components/PropsTable'
import Alert from './Alert'
import cactusTheme from '@repay/cactus-theme'
import { LiveProvider, LiveEditor, LiveError, LivePreview } from 'react-live'

# Alert

The `Alert` component is designed to provide user feedback. The alert is a rectangular shape providing the user a current status based on the icon and background color.

### How it looks

export const code = `<Alert status="success" type="general"> Message</Alert>`

<LiveProvider code={code} scope={{ Alert }}>
  <LiveEditor />
  <LiveError />
  <LivePreview />
</LiveProvider>

## Best practices

The `Alert` component should be used to provide users with feedback based on the user's status. The status can be success, error, warning, and info. The `Alert` component should be set to a general alert or a push notification alert. The general alert is larger in size than the push notifcation. The push notification alert has a button, on the top right side, and an onClick action should be set to handle the click event when implementing a push notification.

## Basic usage

```jsx
import React from 'react'
import { Alert } from '@repay/cactus-web'
...
<Alert status="success" type="general">
  Message
</Alert>
```

## Properties

<PropsTable of={Alert} />
