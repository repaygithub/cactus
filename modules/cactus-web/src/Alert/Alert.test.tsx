import * as React from 'react'
import { cleanup, fireEvent, render } from 'react-testing-library'
import { StyleProvider } from '../StyleProvider/StyleProvider'
import Alert from './Alert'

afterEach(cleanup)

describe('component: Alert', () => {
  test('should render the default props, general info alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert> Message </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render general info alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="info" type="general">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })
  test('should render general error alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="error" type="general">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render general warning alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="warning" type="general">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })
  test('should render general success alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="success" type="general">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render push notification info alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="info" type="push">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })
  test('should render push notification error alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="error" type="push">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render push notification warning alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="warning" type="push">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })
  test('should render push notification success alert', () => {
    const { container } = render(
      <StyleProvider>
        <Alert status="success" type="push">
          Message
        </Alert>
      </StyleProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
