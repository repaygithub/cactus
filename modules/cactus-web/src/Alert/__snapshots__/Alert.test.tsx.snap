// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Alert should render general error alert 1`] = `
.c3 {
  font-size: 24px;
  vertical-align: middle;
}

.c4 .c2 {
  padding-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(353,84%,44%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c2 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(353,84%,44%,0.3);
  border: 2px solid hsl(353,84%,44%);
  width: 100%;
  padding: 24px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c5 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2 c3"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M11.1055728,1.5527864 C11.4740971,0.815737865 12.5259029,0.815737865 12.8944272,1.5527864 L22.8944272,21.5527864 C23.2268777,22.2176875 22.743382,23 22,23 L2,23 C1.25661798,23 0.773122263,22.2176875 1.10557281,21.5527864 L11.1055728,1.5527864 Z M3.61803399,21 L20.381966,21 L12,4.23606798 L3.61803399,21 Z M12,18 C11.4477153,18 11,17.5522847 11,17 C11,16.4477153 11.4477153,16 12,16 C12.5522847,16 13,16.4477153 13,17 C13,17.5522847 12.5522847,18 12,18 Z M13,14 C13,14.5522847 12.5522847,15 12,15 C11.4477153,15 11,14.5522847 11,14 L11,11 C11,10.4477153 11.4477153,10 12,10 C12.5522847,10 13,10.4477153 13,11 L13,14 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render general info alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsl(200,29%,90%);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsl(0,0%,100%);
  border: 2px solid hsl(200,96%,35%);
  width: 100%;
  padding: 24px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render general success alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(145,89%,28%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(145,89%,28%,0.3);
  border: 2px solid hsl(145,89%,28%);
  width: 100%;
  padding: 24px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render general warning alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(47,82%,47%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(47,82%,47%,0.3);
  border: 2px solid hsl(47,82%,47%);
  width: 100%;
  padding: 24px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M1.39339828,12.7071068 C1.00287399,12.3165825 1.00287399,11.6834175 1.39339828,11.2928932 L11.2928932,1.39339828 C11.6834175,1.00287399 12.3165825,1.00287399 12.7071068,1.39339828 L22.6066017,11.2928932 C22.997126,11.6834175 22.997126,12.3165825 22.6066017,12.7071068 L12.7071068,22.6066017 C12.3165825,22.997126 11.6834175,22.997126 11.2928932,22.6066017 L1.39339828,12.7071068 Z M3.51471863,12 L12,20.4852814 L20.4852814,12 L12,3.51471863 L3.51471863,12 Z M12,16 C11.4477153,16 11,15.5522847 11,15 C11,14.4477153 11.4477153,14 12,14 C12.5522847,14 13,14.4477153 13,15 C13,15.5522847 12.5522847,16 12,16 Z M13,12 C13,12.5522847 12.5522847,13 12,13 C11.4477153,13 11,12.5522847 11,12 L11,9 C11,8.44771525 11.4477153,8 12,8 C12.5522847,8 13,8.44771525 13,9 L13,12 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render push notification error alert 1`] = `
.c3 {
  font-size: 24px;
  vertical-align: middle;
}

.c4 .c2 {
  padding-bottom: 4px;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(353,84%,44%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c2 {
  padding-bottom: 4px;
}

.c5 .c2 {
  padding-bottom: 4px;
}

.c6 .c2 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(353,84%,44%,0.3);
  border: 2px solid hsl(353,84%,44%);
  padding: 10px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c7 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2 c3"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M11.1055728,1.5527864 C11.4740971,0.815737865 12.5259029,0.815737865 12.8944272,1.5527864 L22.8944272,21.5527864 C23.2268777,22.2176875 22.743382,23 22,23 L2,23 C1.25661798,23 0.773122263,22.2176875 1.10557281,21.5527864 L11.1055728,1.5527864 Z M3.61803399,21 L20.381966,21 L12,4.23606798 L3.61803399,21 Z M12,18 C11.4477153,18 11,17.5522847 11,17 C11,16.4477153 11.4477153,16 12,16 C12.5522847,16 13,16.4477153 13,17 C13,17.5522847 12.5522847,18 12,18 Z M13,14 C13,14.5522847 12.5522847,15 12,15 C11.4477153,15 11,14.5522847 11,14 L11,11 C11,10.4477153 11.4477153,10 12,10 C12.5522847,10 13,10.4477153 13,11 L13,14 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render push notification info alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsl(200,29%,90%);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsl(0,0%,100%);
  border: 2px solid hsl(200,96%,35%);
  padding: 10px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render push notification success alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(145,89%,28%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(145,89%,28%,0.3);
  border: 2px solid hsl(145,89%,28%);
  padding: 10px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M9.81384674,19.7396292 C9.5249491,19.7396292 9.24415138,19.6325301 9.02725315,19.4381317 L3.38789908,14.3504731 C2.906403,13.9166766 2.86770331,13.1741827 3.30239977,12.6926866 C3.73619624,12.2120905 4.47869019,12.1724908 4.96018627,12.6071873 L9.67974783,16.8641526 L18.9127726,5.4369457 C19.3204693,4.93204981 20.0593633,4.85285046 20.5642592,5.26144713 C21.0682551,5.66914381 21.1465544,6.40803779 20.7388578,6.91293368 L10.7273393,19.3040328 C10.524841,19.5542307 10.2269434,19.7108295 9.90654599,19.7369292 C9.87504624,19.7387292 9.84444649,19.7396292 9.81384674,19.7396292"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render push notification warning alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsla(47,82%,47%,0.3);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsla(47,82%,47%,0.3);
  border: 2px solid hsl(47,82%,47%);
  padding: 10px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M1.39339828,12.7071068 C1.00287399,12.3165825 1.00287399,11.6834175 1.39339828,11.2928932 L11.2928932,1.39339828 C11.6834175,1.00287399 12.3165825,1.00287399 12.7071068,1.39339828 L22.6066017,11.2928932 C22.997126,11.6834175 22.997126,12.3165825 22.6066017,12.7071068 L12.7071068,22.6066017 C12.3165825,22.997126 11.6834175,22.997126 11.2928932,22.6066017 L1.39339828,12.7071068 Z M3.51471863,12 L12,20.4852814 L20.4852814,12 L12,3.51471863 L3.51471863,12 Z M12,16 C11.4477153,16 11,15.5522847 11,15 C11,14.4477153 11.4477153,14 12,14 C12.5522847,14 13,14.4477153 13,15 C13,15.5522847 12.5522847,16 12,16 Z M13,12 C13,12.5522847 12.5522847,13 12,13 C11.4477153,13 11,12.5522847 11,12 L11,9 C11,8.44771525 11.4477153,8 12,8 C12.5522847,8 13,8.44771525 13,9 L13,12 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;

exports[`component: Alert should render the default props, general info alert 1`] = `
.c2 {
  font-size: 24px;
  vertical-align: middle;
}

.c1 {
  box-sizing: border-box;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: hsl(200,29%,90%);
  color: hsl(200,10%,20%);
}

.c1 .c3 {
  padding-bottom: 4px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  background: hsl(0,0%,100%);
  border: 2px solid hsl(200,96%,35%);
  width: 100%;
  padding: 24px 16px;
  border-radius: 8px;
}

.c0 div:first-child {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c0 div:nth-child(2) {
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 8px;
}

.c0 .c4 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  margin-left: auto;
  margin-top: 10px;
}

<div>
  <div
    class="c0"
  >
    <div>
      <div
        class="c1"
      >
        <svg
          class="c2"
          fill="currentcolor"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M22,12 C22,6.47714622 17.5223159,2 12,2 C6.47523326,2 2,6.47577095 2,12 C2,17.524229 6.47523326,22 12,22 C17.5223159,22 22,17.5228538 22,12 Z M24,12 C24,18.6274771 18.6268316,24 12,24 C5.37060996,24 0,18.6287448 0,12 C0,5.37125524 5.37060996,0 12,0 C18.6268316,0 24,5.37252293 24,12 Z M11.057,15.6442233 L11.057,12.5 L9.27493729,12.5 C8.72265254,12.5 8.27493729,12.0522847 8.27493729,11.5 C8.27493729,10.9477153 8.72265254,10.5 9.27493729,10.5 L12.057,10.5 C12.6092847,10.5 13.057,10.9477153 13.057,11.5 L13.057,15.6442233 L15.0505322,15.6442233 C15.602817,15.6442233 16.0505322,16.0919386 16.0505322,16.6442233 C16.0505322,17.1965081 15.602817,17.6442233 15.0505322,17.6442233 L9,17.6442233 C8.44771525,17.6442233 8,17.1965081 8,16.6442233 C8,16.0919386 8.44771525,15.6442233 9,15.6442233 L11.057,15.6442233 Z M11.5,9 C10.6715729,9 10,8.32842712 10,7.5 C10,6.67157288 10.6715729,6 11.5,6 C12.3284271,6 13,6.67157288 13,7.5 C13,8.32842712 12.3284271,9 11.5,9 Z"
          />
        </svg>
      </div>
    </div>
    <div>
      Message
    </div>
  </div>
</div>
`;
